src/network_test.c

```diff
@@ -224,16 +224,17 @@ static int decode_string(char const *in, uint8_t *out, size_t *out_size) {
 }
 
 DEF_TEST(parse_packet) {
-  sockent_t se = {
-      .data.server =
-          (struct sockent_server){
+  sockent_t se
```

Comment:
It is not fully equivalent as `se`  is not nulled and `.data_server` can contain random data.
You can use memset to zero the `se` or try using the multi layer default initializer to circumvent the compilation errors.
Something like:
```
#if HAVE_GCRYPT_H
se = //current code
#else
se = {{}}; // or se = {{{}}} whatever is necessary;
#endif
```

Reply:
ACK. Looking into this, we don't need to differentiate between the with and without gcrypt cases. Simply initializing to `{0}` should be sufficient. The gcrypt specific code only initialized the gcrypt fields to zero, so that can easily be removed.

I pushed an additional commit simplifying that.